Unit13-ObjectOriented/13.1_CarOO/app.js

```diff
@@ -0,0 +1,80 @@
+/*
+Title: Car Object-Oriented Programming Challenge
+Author: Fabricio Fogaroli Ribeiro
+Date: April 7, 2024
+*/
+
+//Global variables
+
+//================================================================
+//Feature Functions
+
+class Vehicle {
+    constructor(make, model, year) {
+        this.make = make;
+        this.model = model;
+        this.year = year;
+    }
+
+    honk() {
+        return "Beep.";
+    }
+
+    toString() {
+        const { make, model, year } = this;
+        return `The vehicle is a ${make} ${model} from ${year}.`;
+    }
+}
+
+class Car extends Vehicle {
+    constructor(make, model, year) {
+        super(make, model, year);
+        this.numWheels = 4;
+    }
+}
+
+class Motorcycle extends Vehicle {
+    constructor(make, model, year) {
+        super(make, model, year);
+        this.numWheels = 2;
+    }
+
+    revEngine() {
+        return `VROOM!!!`;
+    }
+}
+
+class Garage {
+    constructor(size) {
+        this.size = size;
+        this.vehicles = [];
+    }
+
+    add(vehicle) {
+        if (
+            vehicle.make === undefined ||
+            vehicle.model === undefined ||
+            vehicle.year === undefined
+        ) {
```

Comment:
```suggestion
        if (
            !vehicle.make  ||
            !vehicle.model  ||
            !vehicle.year
        ) {
```

Just different ways of writing this, you could event. extract this into a function called isValidVehicle and Use `Object.keys()` to check for required properties

Reply:
Changes the IF statement and added as a separated function. Did not identify a better way using Object.keys().